### DISCUSSION TITLE
How to stop preview window flickering when typing in custom text widget

### DISCUSSION THREAD

**Discussion by WebsiteDons** [elementor/elementor#18562] [Developers Community]
I've added a custom text widget and all works well except each key stroke causes the preview window to flicker as if reloading. It doesn't happen with the core text widget so I am assuming there is an instruction I am missing. I've read through the developer pages but nothing seems to specifically address this issue.

See video of the issue https://www.youtube.com/watch?v=LH0elBgcszU

**Comment by JohnnyHerms** [comment]
I'm also experiencing the same issue.  I think we'd have to implement the content_template() method as per the documentation: https://developers.elementor.com/docs/widgets/widget-rendering/

However, I'm hoping there's a simpler way since replicating all of the code from the render() method into content_template() is quite a lot of duplicate work.  Does anyone know a simpler way?
